### PR TITLE
Selling & Dragging

### PR DESCRIPTION
**Pull request by alexc19113800**
- Selling and store are now mutually exclusive
- Fixed issue that selling does not update cellData
- Dragging is now related linearly to mouse displacement
*Selling is only set up on the SellingTesting scene with the sell button

### CODE REVIEW

File: Space-Zoologist/Assets/Scripts/UI/Store/SellingManager.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+    public void StopSelling() {
+        IsSelling = false;
+    }
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        IsSelling = false;
+        gridSystem = FindObjectOfType<GridSystem>();
```
Comment (tg1230) [inline Space-Zoologist/Assets/Scripts/UI/Store/SellingManager.cs:41]: FindObjectOfType should typically be used if there's no other easy way to get a reference to another object--it's an expensive search and can usually be replaced with better design. It's also fine to use when prototyping and trying to get functionality working, but should be refactored out if possible. In this case, we can just add the GridSystem and TileSystem as serialized fields.

File: Space-Zoologist/Assets/Scripts/UI/Store/SellingManager.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            {
+                cellData = gridSystem.CellGrid[cellPos.x, cellPos.y];
+            }
+            else
+            {
+                // Debug.Log($"Grid location selected was out of bounds @ {cellPos}");
+                return;
+            }
+
+            if (cellData.ContainsMachine)
```
Comment (tg1230) [inline Space-Zoologist/Assets/Scripts/UI/Store/SellingManager.cs:78]: Break up the logic in these if statements into more functions. We want our code to be easy to read so it's easier to maintain and work with.

**Review by tg1230 — CHANGES_REQUESTED** [review]
Two small fixes needed before we can merge: refactor out all uses of FindObjectByType, and break up the if statements in Update in the SellingManager into more functions.

File: Space-Zoologist/Assets/Scripts/UI/Store/SellingManager.cs (hunk 1 above)
Comment (alexc19113800) [reply Space-Zoologist/Assets/Scripts/UI/Store/SellingManager.cs]: Addressed.

File: Space-Zoologist/Assets/Scripts/UI/Store/SellingManager.cs (hunk 2 above)
Comment (alexc19113800) [reply Space-Zoologist/Assets/Scripts/UI/Store/SellingManager.cs:80]: I split it into SellMachineOnCell(), SellFoodOnCell(), SellWallOnCell().

**Pull request merged**